hail/python/hailtop/batch/backend.py

```diff
@@ -309,18 +309,51 @@ class ServiceBackend(Backend):
     >>> b.run() # doctest: +SKIP
     >>> service_backend.close()
 
+    Instead of a bucket, a full path may be specified for the remote temporary directory:
+
+    >>> service_backend = ServiceBackend('my-billing-account',
+    ...                                  remote_tmpdir='gs://my-bucket/temporary-files/')
+    >>> b = Batch(backend=service_backend)
+    >>> b.run() # doctest: +SKIP
+    >>> service_backend.close()
+
     Parameters
     ----------
     billing_project:
         Name of billing project to use.
     bucket:
-        Name of bucket to use.  Should not include the ``gs://``
-        prefix.
+        Name of bucket to use. Should not include the ``gs://`` prefix. Cannot be used with
+        remote_tmpdir. Temporary data will be stored in the "/batch" folder of this
+        bucket. Using this parameter as a positional argument is deprecated.
+    remote_tmpdir:
+        Temporary data will be stored in this google cloud storage folder. Cannot be used with bucket. Example: "gs://
+
     """
 
     def __init__(self,
-                 billing_project: str = None,
-                 bucket: str = None):
+                 *args,
+                 bucket: Optional[str] = None,
+                 remote_tmpdir: Optional[str] = None
+                 ):
+        billing_project: Optional[str]
+        if len(args) > 2:
```

Comment:
I'm worried about the backwards compatibility of this change. This seems complicated. Why not just keep the interface the way it was before? Otherwise, the right interface seems to be `def __init__(self, *, billing_project=None, bucket = None, remote_tmpdir = None)`